src/figuresClasses.ts

```diff
@@ -1,19 +1,105 @@
-export interface Figure {
-
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = Shape.Triangle;
+
+    if (!Triangle.isRightTriangle(a, b, c)) {
+      this.throwError();
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  getArea() {
+    const halfP = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(
+      halfP * (halfP - this.a) * (halfP - this.b) * (halfP - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+
+  static isRightTriangle(a: number, b: number, c: number): boolean {
+    const isRight = !((a >= b + c) || (b >= a + c) || (c >= a + b));
+    const isAllPositive = a && b && c;
+
+    return Boolean(isRight && isAllPositive);
+  }
 
+  throwError(): void {
+    throw Error(`${this.shape} not right`);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Shape;
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.shape = Shape.Circle;
```

Comment:
You can assign this value to your shape immediately
Here and in other figures.
```suggestion
  public shape: Shape = Shape.Circle;

  constructor(
    public color: Color,
    public radius: number,
  ) {
```